website/blog/2023-08-08-deploy-run-ferretdb-leafcloud.md

```diff
@@ -0,0 +1,443 @@
+
+```sh
+openstack server list
+```
+
+If you get an authentication error, try to run `source ~/leafcloudopenrc.sh` again and enter the right password for your account.
+
+Leafcloud provides a set of cluster templates that you can take advantage of right away.
+See the templates by running:
+
+```sh
+openstack coe cluster template list
+```
+
+Select the best template for your project.
+
+We'll be using the `K8s-ha-v1.21.2-template-v2.0-rc3` template for the guide since it comes with Kubernetes OpenStack autoscaling, encrypted cinder volumes for containers and hosts (persistent volume claims), and high-availability load balancers.
+
+Using the selected template, we can now create a new cluster.
+The keypair, which we associate using the `--keypair` parameter, will be integrated into the hosts, granting us root SSH access (with 'core' as the default user).
+The keypairs are essential for encrypting SSH traffic for your instances and enabling secure communications.
+
+To create your cluster, use this command:
+
+```sh
+openstack coe cluster create my-k8s-cluster --cluster-template k8s-ha-v1.21.2-template-v2.0-rc3 --keypair <keypair>
+```
+
+You may need to wait a few minutes for this process to be complete.
+To check up on the installation enter the following:
+
+```sh
+openstack coe cluster list
+```
+
+The installation is complete once the status of the cluster changes from CREATE_IN_PROGRESS to CREATE_COMPLETE.
+
+Now we need to fetch the configuration file for the cluster.
+Do this by running the following:
+
+```sh
+openstack coe cluster config my-k8s-cluster
+```
+
+A file named 'config' will be downloaded to your home directory.
+
+```sh
+export KUBECONFIG=/home/<username>/config
+```
+
+Your cluster should now be reachable using Kubectl.
+Enter the command to confirm:
+
+```sh
+kubectl get nodes -o wide
+```
+
+Lastly, we need to set up the storage class to use persistent volume claims (PVCs).
+Do this by creating the storageclass.yaml file:
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: cinder-csi
+  annotations:
+    storageclass.kubernetes.io/is-default-class: 'true'
+provisioner: cinder.csi.openstack.org
+```
+
+You can apply it by running:
+
+```sh
+kubectl apply -f storageclass.yaml
+```
+
+### Creating the Postgres Cluster
+
+Use kustomize postgres-operator 5.4.0 to install the postgres operator.
+
+#### Installing the Postgres Operator
+
+In this section, we'll be creating the PostgreSQL cluster.
+We'll start by installing the latest version of the PostgreSQL operator, which is essential when setting up extremely reliable PostgreSQL clusters on Kubernetes.
+
+Here we will use kustomize to install the operator, but first, we need to fork the GitHub PostgreSQL Operator website.
+
+```text
+YOUR_GITHUB_UN="<your GitHub username>"
+git clone --depth 1 "git@github.com:${YOUR_GITHUB_UN}/postgres-operator-examples.git"
+cd postgres-operator-examples
+```
+
```

Comment:
Maybe we should comment here that one can modify the cloned files according to their needs. We use the default configs, but other people might want to configure their cluster in a particular way.